src/main/java/com/poortorich/security/filter/auth/JwtAuthenticationFilter.java

```diff
@@ -0,0 +1,134 @@
+package com.poortorich.security.filter.auth;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.poortorich.auth.jwt.util.JwtCookieManager;
+import com.poortorich.auth.jwt.util.JwtTokenExtractor;
+import com.poortorich.auth.jwt.validator.JwtTokenValidator;
+import com.poortorich.auth.response.enums.AuthResponse;
+import com.poortorich.global.exceptions.UnauthorizedException;
+import com.poortorich.global.response.BaseResponse;
+import com.poortorich.global.response.Response;
+import com.poortorich.security.constants.SecurityConstants;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.MediaType;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    private final JwtTokenValidator tokenValidator;
+    private final JwtTokenExtractor tokenExtractor;
+    private final JwtCookieManager cookieManager;
+    private final UserDetailsService userDetailsService;
+
+    @Override
+    protected void doFilterInternal(
+            HttpServletRequest request,
+            HttpServletResponse response,
+            FilterChain filterChain
+    ) throws ServletException, IOException {
+
+        if (Arrays.asList(SecurityConstants.PERMIT_ALL_ENDPOINTS).contains(request.getRequestURI())) {
+            filterChain.doFilter(request, response);
+            return;
+        }
+
+        try {
+            if (!processAccessToken(request, response)) {
+                processRefreshToken(request, response);
+            }
+        } catch (UnauthorizedException exception) {
+            cookieManager.clearAuthCookie(response);
+            setResponseMessage(response, exception.getResponse());
+            return;
+        }
+        filterChain.doFilter(request, response);
```

Comment:
수직 공백을 추가하는 게 더 가독성이 좋을까요? 🤔
```suggestion
        }
        
        filterChain.doFilter(request, response);
```